src/System.Net.Sockets/src/System/Net/Sockets/IOControlKeepAlive.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Net.Sockets
+{
+    internal class IOControlKeepAlive
+    {
+        private const uint OnOff = 1u;
+        private const uint DefaultTimeMs = 7200000u;
+        private const uint DefaultIntervalMs = 1000u;
+        private static readonly ConditionalWeakTable<Socket, IOControlKeepAlive> SocketKeepAliveCwt;
+        private static readonly int UIntSize;
+
+        private uint timeMs;
+        private uint intervalMs;
+
+        static IOControlKeepAlive()
+        {
+            SocketKeepAliveCwt = new ConditionalWeakTable<Socket, IOControlKeepAlive>();
+            UIntSize = Marshal.SizeOf<uint>();
+        }
```

Comment:
It'd be better to just initialize thisas part of the declaration, e.g.
```C#
private static readonly ConditionalWeakTable<Socket, IOControlKeepAlive> s_ocketKeepAliveCwt = new ConditionalWeakTable<Socket, IOControlKeepAlive>();
```
Adding an explicit static cctor prevents the C# compiler from marking the class as beforefieldinit, which has perf implications.

Reply:
`beforefieldinit` gives the JIT more freedom as to if/when it does the initialization.  Without `beforefieldinit`, the code generated by the JIT needs to ensure that the static cctor is run once and only once the first time the type is used, which means it needs to generate code that guards such accesses to ensure the static cctor has been run and to run it if it hasn't.  With `beforefieldinit`, the JIT could choose to, for example, do the initialization at JIT time, which means it doesn't have to output such checks.

Reply:
Thank's!